src/transport/PeerMessageCounter.h

```diff
@@ -118,7 +135,7 @@ class PeerMessageCounter
             return CHIP_NO_ERROR;
         case Status::Synced: {
             CHIP_ERROR err = Verify(counter);
-            if (err == CHIP_ERROR_MESSAGE_COUNTER_OUT_OF_WINDOW)
+            if (err == CHIP_ERROR_MESSAGE_COUNTER_OUT_OF_WINDOW && !mIsGroup)
```

Comment:
Again, I suspect it might be better to rename this to VerifyUnencrypted and have a VerifyGroupTrustFirst that calls the group version of Verify, whatever we call it.

Should be doable, because at all the verification callsites we know exactly what sort of rules we want to apply....

Reply:
For nomenclature reasons, I added a default argument `useGroupAlgorithm=false` to this method